### ISSUE TITLE
Math.random returns 0 on the sprig

### ISSUE THREAD

**Issue by peyeyey** [hackclub/sprig#730]
I have a `Math.random` running in my game.  It works fine in the browser, but on the sprig, it always returns 0.  Try running the following code on the browser and sprig:

```
/*
@title: random_test
@author: peyeyey
*/

setInterval(function() {
  clearText();
  addText(`${Math.random()}`, { x: 2, y: 8, color: color`L`});
}, 500);

const player = "p";

setLegend(
  [ player, bitmap`
................
................
.......000......
.......0.0......
......0..0......
......0...0.0...
....0003.30.0...
....0.0...000...
....0.05550.....
......0...0.....
.....0....0.....
.....0...0......
......000.......
......0.0.......
.....00.00......
................`]
);

let level = 0;
const levels = [
  map`
p.
..`,
];

setMap(levels[level]);

```

**Comment by pkaminski** [comment]
We tested this with a version of `pico-os.uf2` downloaded on Jan 13th, so it includes commit [35232b5](https://github.com/hackclub/sprig/commit/35232b5c99d63cb95ac7a1450237f6ee93e380ff), which presumably incorporates @kognise's fix to `spade`'s `Math.random` in [1410545](https://github.com/hackclub/spade/commit/141054559a49eca6c66f4aa76dde34f436a9597c).  It doesn't appear to be working, however.

**Comment by Vishram1123** [comment]
Developing scenario: It appears as though the Date.now(); function isn't working - 


[datenow.webm](https://user-images.githubusercontent.com/64966832/213925316-3001efdc-9598-48a3-ae28-2e1a2869930a.webm)

Code: 

```js
/*
@title: time_test
@author: vishram1123
*/



const level =  map`
.`;

setMap(level);
setInterval(() => {
  addText(Date.now() + "", {
        x: 4,
        y: 7,
        color: color`0`
      });
}, 500);
```

**Comment by Vishram1123** [comment]
There appears to be a [pull request to resolve this issue on the spade repo](https://github.com/hackclub/spade/pull/5)

**Comment by kognise** [comment]
The base problem here is that the Raspberry Pi Pico doesn't have a battery backed clock, so every time the device is power cycled the time gets reset to the same value. This means this isn't actually a proper source of randomness. In my stupidity I only tested on desktop Spade, where Date.now() was different each time, and it worked, so I was happy.

We should use a different entropy source, maybe something temperature-derived if there are sensors for such. [Possible example here](https://forums.raspberrypi.com/viewtopic.php?t=302960)

**Comment by Vishram1123** [comment]
Maybe as a workaround for now, you can set the seed to Date.now() + 1, such that on desktop it would actually use the current time + 1 as the seed, but on the sprig, it would use 1 as a seed (which provides "randomness" to an extent, instead of just being zero all of the time)

**Comment by devramsean0** [comment]
The pico has a temperature sensor built in

**Comment by leomcelroy** [comment]
I was thinking you could seed it with an analog reading of an unconnected gpio pin.

**Comment by kognise** [comment]
Fixed in 257119236aa4f15509d56ac82cc0cb4dddd9d3a7! Seeded with 5 GPIO pin reads (including temp sensor) on hardware, and date on PC.
